pkg/inputs/snmp/x/meraki/meraki.go

```diff
@@ -0,0 +1,685 @@
+	SignalType     string     `json:"signalType"`
+	Usage          []uplinkInterfaceUsage
+	LatencyLoss    deviceUplinkLatency
+}
+
+func (du *deviceUplink) SetLatencyLoss(u deviceUplinkLatency) {
+	found := false
+	for i, ul := range du.Uplinks {
+		if ul.Interface == u.Uplink {
+			du.Uplinks[i].LatencyLoss = u
+			found = true
+		}
+	}
+	if !found {
+		du.Uplinks = append(du.Uplinks, uplink{
+			Interface:   u.Uplink,
+			LatencyLoss: u,
+		})
+	}
+}
+
+func (du *deviceUplink) SetUsage(uplinkHistories []uplinkUsage) {
+	for _, ts := range uplinkHistories {
+		for _, u := range ts.ByInterface {
+			found := false
+			for i, ul := range du.Uplinks {
+				if ul.Interface == u.Interface {
+					if du.Uplinks[i].Usage == nil {
+						du.Uplinks[i].Usage = make([]uplinkInterfaceUsage, 0)
+					}
+					du.Uplinks[i].Usage = append(du.Uplinks[i].Usage, u)
+					found = true
+				}
+			}
+			if !found {
+				du.Uplinks = append(du.Uplinks, uplink{
+					Interface: u.Interface,
+					Usage: []uplinkInterfaceUsage{
+						u,
+					},
+				})
+			}
+		}
+	}
+}
+
+type deviceUplink struct {
+	Serial         string    `json:"serial"`
+	Model          string    `json:"model"`
+	LastReportedAt time.Time `json:"lastReportedAt"`
+	Uplinks        []uplink  `json:"uplinks"`
+	NetworkID      string    `json:"networkId"`
+	network        *networkDesc
+}
+
+/**
+1) Get all the uplinks with status
+2) Get the latency for these uplinks for any which have them
+3) Get usage for each network.
+
+*/
+
+func (c *MerakiClient) getUplinks(dur time.Duration) ([]*kt.JCHF, error) {
+
+	uplinkSet := map[string]*deviceUplink{}
+	for _, org := range c.orgs {
+		params := organizations.NewGetOrganizationUplinksStatusesParams()
+		params.SetOrganizationID(org.ID)
+
+		prod, err := c.client.Organizations.GetOrganizationUplinksStatuses(params, c.auth)
+		if err != nil {
+			return nil, err
+		}
+
+		b, err := json.Marshal(prod.GetPayload())
+		if err != nil {
+			return nil, err
+		}
+
+		var uplinks []deviceUplink
+		err = json.Unmarshal(b, &uplinks)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, uplink := range uplinks {
+			for _, network := range c.networks {
+				if network.ID == uplink.NetworkID {
+					lnet := network
+					uplink.network = &lnet
+				}
+			}
+			if uplink.network == nil {
+				c.log.Errorf("Missing Network for Uplink %s", uplink.NetworkID, uplink.Serial)
+			}
+			if _, ok := uplinkSet[uplink.Serial]; ok {
+				c.log.Errorf("Duplicate Uplink %s", uplink.Serial)
+			} else {
+				uplinkSet[uplink.Serial] = &uplink
+			}
+		}
+	}
+
+	err := c.getUplinkUsage(dur, uplinkSet)
+	if err != nil {
+		return nil, err
+	}
+
+	// Now, load latency for any of these which have them:
+	err = c.getUplinkLatencyLoss(dur, uplinkSet)
+	if err != nil {
+		return nil, err
+	}
+
+	//c.log.Infof("Loaded %d Uplinks for %d Organizations", len(uplinkSet), len(c.orgs))
+
+	return c.parseUplinks(uplinkSet)
+}
+
+type uplinkTS struct {
+	TS          time.Time `json:"ts":`
```

Comment:
typo?  should it be `json:"ts"` (no extra colon).

Reply:
good catch!